app/code/core/Mage/Core/Model/Resource/Db/Abstract.php

```diff
@@ -656,10 +657,10 @@ protected function _checkUnique(Mage_Core_Model_Abstract $object)
 
                 if (is_array($unique['field'])) {
                     foreach ($unique['field'] as $field) {
-                        $select->where($field . '=?', trim($data->getData($field)));
+                        $select->where($field . '=?', trim((string)$data->getData($field)));
                     }
                 } else {
-                    $select->where($unique['field'] . '=?', trim($data->getData($unique['field'])));
+                    $select->where($unique['field'] . '=?', trim((string)$data->getData($unique['field'])));
```

Comment:
```suggestion
                } elseif ($data->getData($unique['field']) !== null) {
                    $select->where($unique['field'] . '=?', trim($data->getData($unique['field'])));
```

Reply:
This is working when I save an attributes set, but are there side effects? I don't know.